src/sqlfluff/core/rules/std/L015.py

```diff
@@ -30,10 +30,42 @@ def _eval(self, segment, raw_stack, **kwargs):
 
         Look for DISTINCT keyword immediately followed by open parenthesis.
         """
-        # We only trigger on start_bracket (open parenthesis)
-        if segment.name == "start_bracket":
-            filt_raw_stack = self.filter_meta(raw_stack)
-            if len(filt_raw_stack) > 0 and filt_raw_stack[-1].name == "DISTINCT":
-                # If we find DISTINCT followed by open_bracket, then bad.
-                return LintResult(anchor=segment)
-        return LintResult()
+        # We only trigger on expressions that begin with start_bracket (open
+        # parenthesis)
+        if segment.type == "expression":
+            segments_filt = self.filter_meta(segment.segments)
+            if (
+                len(segments_filt)
+                and segments_filt[0].type == "start_bracket"
+                and segments_filt[0].raw == "("
+            ):
+                filt_raw_stack = self.filter_meta(raw_stack)
+                if len(filt_raw_stack) and filt_raw_stack[-1].name == "DISTINCT":
+                    # If we find open_bracket immediately following DISTINCT,
+                    # then bad.
+                    fixes = []
+                    if len(segments_filt) >= 2:
```

Comment:
Could you add a comment to explain the significant of >= 2? An inline example of what would be caught and what wouldn't be would be helpful for me to understand!

Reply:
Will do. In looking at this, I found and fixed a subtle bug... in the code just below, where I create a `LintFix`, I was accidentally using `segment.segments` rather than `segments_filtered`. Fixing this bug makes the reason for the length check much clearer, because now we're essentially doing:

```
if len(segments_filtered) >= 2:
    ...
    self.filter_meta(segments_filtered)[1],
    ...
```

so the `if` check prevents accessing the second element in a list that is too short.

TBH, I had wondered if there was a better way to do this, for example placing the whitespace after the open parenthesis, but that wasn't working for me.

I can see that placing something after a deleted element is a bit odd, although it seems like it should work. 🤷

Reply:
After studying the code some more, I found a much cleaner way to refactor it that no longer requires this kind of length check. Now it only needs to check for non-zero length **after** the deletions, which I think is easier to understand.